db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1551,13 +1551,38 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
     if (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load()) {
       s = SwitchMemtable(cfd, &context);
     }
-
     if (s.ok()) {
       if (cfd->imm()->NumNotFlushed() != 0 || !cfd->mem()->IsEmpty() ||
           !cached_recoverable_state_empty_.load()) {
         flush_memtable_id = cfd->imm()->GetLatestMemTableID();
         flush_req.emplace_back(cfd, flush_memtable_id);
       }
+      if (immutable_db_options_.persist_stats_to_disk) {
```

Comment:
Do we want to flush the stats cf when an automatic flush is triggered?

Reply:
IIRC `FlushMemTable` handles manual flush, changes in `ScheduleFlushes` takes care of automated flush.

Reply:
Probably check `SwitchWAL` and `HandleWriteBufferFull` where auto-flush can trigger?

Reply:
added `MaybeFlushStatsCF` to all three callers of `SwitchMemtable` in db_impl_write.cc